File: credit.c
#include <cs50.h>
#include <stdio.h>

int digitsum(int n){
    int m=0;
    while (n!=0)
    {
        m += n % 10;
        n /= 10;
    }
    return m;
}
int poweroftwo(bool n){
    return n ? 2 : 1;
}
string what_is_it(long m){
    int count = 0;
    string result;
    while (m >= 100)
    {
        count++;
        m /=10;
    }
    count +=2;
    if (count == 15 && (m == 34 || m == 37))
    {
        result = "AMERICAN EXPRESS";
    }
    else if (count == 16 && (m == 51 || m == 52 || m == 53 || m == 54 || m == 55))
    {
        result = "MASTERCARD";
    }
    else if ((count == 13 || count == 16) && (m / 10) == 4){
        result = "VISA";
    }
    else 
    {
        result = "INVALID";
    }
    return result;
}
int main(void)
{
    long n;
    bool m=0;
    int z = 0;
    string result;
    do
    {
        n = get_long("Number: ");
    }
    while (!(n > 0));
    result = what_is_it(n);
    while (n > 0)
    {
        z += digitsum(n % 10 * poweroftwo(m));
        n /= 10;
        m = !m;
    }
    if (z % 10 == 0)
    {
        printf("%s\n" ,result);
    }
    else 
    {
        printf("INVALID\n");
    }
    return 0;
}

File: credit.py
def digitsum(n):
    a = 0
    while n:
        a += n % 10
        n //= 10
    return a

n = input("Number: ")
i = False
sum = 0
for j in reversed(n):
    sum += digitsum(2**i * int(j))
    i = i == 0
if (sum % 10 != 0):
    print("INVALID")
elif ((n[0:2] == "34" or n[0:2] == "37") and len(n) == 15):
    print("AMEX")
elif ((n[0:2] == "51" or n[0:2] == "52" or n[0:2] == "53" or n[0:2] == "54" or n[0:2] == "55") and len(n) == 16):
    print("MASTERCARD")
elif ((n[0] == "4") and (len(n) == 13 or len(n) == 16)):
    print("VISA\n")
else:
    print("INVALID")